README.md

```diff
@@ -1,2 +1,300 @@
+
+Attaches a wakeup interrupt to digital pin `pin`. `callback` callback is the function to be called when the wakeup interrupt occurs. 
+
+*Parameters*
+
+none
+
+*Return value*
+
+none
+
+### ``CuriePower.dettachInterruptWakeup(uint32_t pin)``
+
+```
+void CuriePower.dettachInterruptWakeup(uint32_t pin)
+```
+
+Removes any wakeup interrupt attached to digital pin `pin`. 
+
+*Parameters*
+
+none
+
+*Return value*
+
+none
+
+# Tutorials
+
+## Tutorial #1: TimedWakeup Example
+
+This sketch demonstrates the simplest way to use the CuriePower library. It blinks the LED a few times, goes to sleep for a certain amount then goes back at the start of loop()
+
+```cpp
+#include <Power.h>
+
+void setup() {
+  pinMode(LED_BUILTIN, OUTPUT);
+}
+
+void loop() {
+  for(int i = 0; i < 5; i++)
+  { 
+    digitalWrite(LED_BUILTIN, HIGH);
+    delay(100);
+    digitalWrite(LED_BUILTIN, LOW);
+    delay(100);
+  }
+  PM.sleep(1000);
+}
+```
+
+The line of code that is most interesting here is:
+```cpp
+PM.sleep(1000);
+```
+This puts the SoC into sleep, drawing significantly less power, for 1000ms or 1s.
+
+In what situations is this most useful?
+
+Lets says you have a battery powered project that reads a sensor value once every second and saves it to an SD card.
+Simply using delay() will work but you will notice that you will run out of battery pretty fast. That is becasue even though the code is not doing much inside delay, it is still running everything at full clock speed and all peripeherals are turned on.
+When we put the SoC to sleep, several things are turned off. This includes most of the peripherals. voltage rails, and some clocks. Basically, it draws much less power and no code is running until a wake interrupt is generated.
+
+Many Arduino projects typically have a loop where you read a sensor, do something with that reading, and then delaying for a set amount of time.
+In most cases, reading the sensor and doing something with that reading takes very little time, much smaller than the delay duration. This means that we are wasting a lot of power inside the delay doing nothing.
+By placing the SoC to sleep instead of just waiting inside the delay, we can save a considerable amount of power in most applications.
+
+
+## Tutorial #2: WakeFromIMU Example
+
+This sketch uses CuriePower library and the CurieIMU library together and demonstrates the use of an interrupt to wake the Curie SoC from sleep.
+
+```cpp
+#include <Power.h>
+#include "CurieIMU.h"
+
+
+void setup() {
+  pinMode(LED_BUILTIN, OUTPUT);
+  CurieIMU.begin();
+  CurieIMU.attachInterrupt(wakeup);
+  CurieIMU.setDetectionThreshold(CURIE_IMU_MOTION, 20);      // 100mg
+  CurieIMU.setDetectionDuration(CURIE_IMU_MOTION, 10);       // trigger times of consecutive slope data points
+  CurieIMU.interrupts(CURIE_IMU_MOTION);
+  
+}
+
+void loop() {
+  PM.sleep();
+  digitalWrite(LED_BUILTIN, HIGH);
+  delay(3000);
+  digitalWrite(LED_BUILTIN, LOW);
+}
+
+void wakeup()
+{
+  PM.wakeFromDoze();
```

Comment:
they will have to call it themselves for this specific example because the ISR is setup by the CurieIMU library.